sysconfig/network-scripts/ifdown-post

```
@@ -49,9 +49,12 @@ else
 fi
 fi
 
-# Inform firewall
-if [ -x /usr/bin/firewall-cmd -a "${REALDEVICE}" != "lo" ]; then
-    /usr/bin/firewall-cmd --remove-interface="${DEVICE}" > /dev/null 2>&1
+# Reset firewall ZONE to "default":
+if [ "${REALDEVICE}" != "lo" ]; then
+    dbus-send --system --dest=org.fedoraproject.FirewallD1 \
```

Unless you specify --print-reply this will be fire and forget. If you care about the result/exit code then you must specify --print-reply.

---

Yes, I realize that this will be *fire and forget*, and IMHO that was the point of the BZ in the first place -- they need a speedup for the network-scripts, and waiting for the reply & exit-code would not be so fast. (Correct me if I'm wrong.)

And lastly, we didn't check for the exit code before either, and after discussion with our colleagues we do not plan to check for it now.

---

In relative numbers, --print-reply adds quite a lot of time to the call. However, IMO compared to the original firewall-cmd it seems insignificant. I leave it to you if you think the error output would be worth it.

firewall-cmd --zone="${ZONE}" --change-interface="${DEVICE}"
> 0.309s

dbus-send --system --dest=org.fedoraproject.FirewallD1 /org/fedoraproject/FirewallD1 org.fedoraproject.FirewallD1.zone.changeZoneOfInterface string:"" string:"dummy_1"
> 0.005s

dbus-send --print-reply --system --dest=org.fedoraproject.FirewallD1 /org/fedoraproject/FirewallD1 org.fedoraproject.FirewallD1.zone.changeZoneOfInterface string:"" string:"dummy_1"
> 0.010s

---

We are actually not sure right now, what consequences the checking for exit codes / replies could have, especially in RHV environment with hundreds to thousands of interfaces. So far, we have been "successfully" avoiding the exit codes / replies, so in order to keep things working as they were before - especially in this case - we will keep silently ignoring the exit codes / replies.

---

Another thing worth pointing out is that this changes a synchronous call into an asynchronous call - it no longer blocks here. Any following commands or actions that depend on this command having already taken effect may be racey.

---

@erig0, that's actually a good point. There's one place where we might need to wait for the reply (i.e. synchronous call), and that's the IPv6 initialization. I will update the commit accordingly. :)